### ISSUE TITLE
Convergence warnings for the quadratic problem

### ISSUE THREAD

**Issue by Marius1311** [theislab/moscot#441]
I'm on the latest commits for both OTT (`0.3.2.dev3+g3ebac63`) and moscot (`0.1.1.dev27+gce367a4`). 

I'm trying to solve a fairly basic FGW problem using the generic FGW solver. I'm setting up and solving my problem as follows:
```python
fgwp = FGWProblem(data)
fgwp = fgwp.prepare(key="day", joint_attr="X_LSI", GW_x="X_LSI", GW_y="X_LSI")
fgwp = fgwp.solve(epsilon=0.1, scale_cost="mean", alpha=0.5)
```
So this is balanced, entropically regularized. The problem contains 3 time points, and both FGW problems do not converge (see Fig. 1 below). The problem is, it's a bit hard to diagnose why these problems do not converge because the warning just says that it did not, but I can see that the outer iteration budget hasn't been exceeded, so it must either be the internal linear solver failing or some numerical issues somewhere. I noticed that it always stops exactly once it reaches `min_iterations`, and then it says it did not converge. 

It would thus be nice if the convergence warning would be a bit more verbose and would tell me whether the outer or inner iteration budget was exceeded, or whether it encountered any other difficulties. That would make it much easier to investigate the problem. 

## Fig. 1
![Screenshot 2023-01-03 at 13 55 16](https://user-images.githubusercontent.com/24494650/210361304-c432839f-545a-4396-83b2-a0e3cdb72a87.png)

**Comment by MUCDK** [comment]
Any chance you can provide a brief example to reproduce, e.g. direct to where to load the data from?

**Comment by Marius1311** [comment]
yes, I send you the dataset location on our server in mattermost

**Comment by Marius1311** [comment]
This particular case might be a consequence of outliers in the cost matrix - I checked and found, for each time point, a small number of cells that's really far away from the other cells. Removing these cells, and running the same commends (below) yields the convergend solutions of Fig. 1 below

## Fig. 1
![Screenshot 2023-01-03 at 18 04 35](https://user-images.githubusercontent.com/24494650/210405288-b374d633-feb0-422c-b7be-ae7b46c62a80.png)

**Comment by michalk8** [comment]
closed due to inactivity
